[docs] Add organization token (#704)
grafana/k6-docs: src/data/markdown/docs/03 cloud/04 Integrations/04 Token.md

excerpt: 'How to authenticate with k6 Cloud token'
---

You'll need to authenticate to use k6 Cloud, whether it's for streaming results or running tests in the cloud. Your API token enables the interaction with k6 Cloud using the k6 CLI or through the REST API.

> In k6 Cloud there are two ways how you can generate an API Token, via account or organization settings. These tokens differ in the kind of access they provide - [account level API token](#account-api-token) will grant access through your account with k6, while [organization level API token](#organization-api-token) will grant organization-wide access for actions not tied to a user.    

Reviewer: ```suggestion
You can generate in API token for k6 Cloud in two ways, each with different levels of access:
- [Account-level API tokens](#account-api-token) grant access through your account with k6
- [Organization-level API tokens](#organization-api-token) grant organization-wide access for actions not tied to a user.    
```

Reviewer 2: > You can generate in API token for k6 Cloud in two ways, each with different levels of access:

I might be wrong as a non-native speaker, but a `level of access` sounds to me like with one token you get more permissions than with the other which is not necessarily true. It's just that there are different usages for the two tokens and the authentication happens on a different level, either through a user where the user's permissions are checked or globally for the organisation (than an admin must have created and provided the token).

Reviewer: You're completely right! I changed it again using the terms you just used:

> You can generate two types of API tokens for k6 Cloud, depending on how you want to authenticate:

> Below you'll find instructions on how to generate API tokens and examples on how to utilize them.

Reviewer: ```suggestion
To generate API tokens, you can follow these steps or adapt these examples.
```

> #### Docker Users
>
> If you're running k6 in a Docker container, make sure that the k6 config file where the k6 Cloud API authentication information is stored is persisted via a Docker volume to the host machine, using the `-c/--config PATH/TO/CONFIG_FILE` CLI flag. Example:
```bash
docker run --rm -i -v /path/on-host:/path/in-container/ grafana/k6 login cloud -c /path/in-container/config.json
```

> #### Integrating with CI
>
> If you are integrating k6 into your CI pipeline, we recommend using an [organization level API token](#organization-api-token) to authenticate and not exposing your username/password within your CI configuration files or as variables. Alternatively you can use an [account level API token](#account-api-token) with a user created specially for the purpose of running CI tests.

## Account API token
An account-level API token provides API access to your account with k6. To generate the token, go to **Account settings** and select **API token**.
From there you can copy, see, and regenerate the token.

![account token view](./images/04-Token/account-api-token-view.png)

## Organization API token

> An organization API token provides organization-wide API access with k6. It is intended for actions that are not tied to a specific user like running tests from a CI platform. To generate a token, go to **Organization settings** and select **API token**.

Reviewer: Made a slight guess in this edit. Please check before committing.

> From there you can create, see, and regenerate the tokens. Maximum amount of tokens an organization can create by default is 5.

Reviewer: ```suggestion
From there you can create, see, and regenerate the tokens.
The maximum amount of tokens that an organization can create is 5 by default.
```

> Note that when using an organization API token you must always specify the project the test runs will be created under. Learn how to set the `Project ID` [here](https://k6.io/docs/cloud/creating-and-running-a-test/cloud-tests-from-the-cli/#running-tests-under-a-different-project-than-your-default-one). 

Reviewer: ```suggestion
Note that when using an organization API token, you must specify the project where the test runs will be created.
Read about [how to set the `Project ID`](https://k6.io/docs/cloud/creating-and-running-a-test/cloud-tests-from-the-cli/#running-tests-under-a-different-project-than-your-default-one). 
```

excerpt: 'How to authenticate with k6 Cloud token'
---

You'll need to authenticate to use k6 Cloud, whether it's for streaming results or running tests in the cloud. Your API token enables the interaction with k6 Cloud using the k6 CLI or through the REST API.

In k6 Cloud there are two ways how you can generate an API Token, via account or organization settings. These tokens differ in the kind of access they provide - [account level API token](#account-api-token) will grant access to your account with k6, while [organization level API token](#organization-api-token) will grant access to your organization.    

Below you'll find instructions on how to generate API tokens and examples on how to utilize them.

> #### Google/GitHub Single Sign-On Users
>
> For Single Sign-On (SSO) users, `k6 login cloud` requires a k6 Cloud account with an email and a password. If you don't want to create a password (using [Forgot Password](https://app.k6.io/account/forgot)), you can authenticate via API token. See [Authenticating with API token](#authenticating-with-api-token) for more information.

> #### Docker Users
>

> > If you're running k6 in a Docker container you'll need to make sure that the k6 config file where the k6 Cloud API authentication information is stored to is persisted via a Docker volume to the host machine, using the `-c/--config PATH/TO/CONFIG_FILE` CLI flag. Example:

Reviewer: ```suggestion
> If you're running k6 in a Docker container, make sure that the k6 config file where the k6 Cloud API authentication information is stored is persisted via a Docker volume to the host machine, using the `-c/--config PATH/TO/CONFIG_FILE` CLI flag. Example:
```

```bash
docker run --rm -i -v /path/on-host:/path/in-container/ grafana/k6 login cloud -c /path/in-container/config.json
```

> #### Integrating with CI
>
> If you are integrating k6 into your CI pipeline, we recommend using one of the token methods to authenticate and not exposing your username/password within your CI configuration files or as variables.

## Account API token

> Account level API token provides API access to your account with k6. To generate the token select **API token** menu item under **Account settings**, where you'll be able to copy, see and regenerate the token.

Reviewer: ```suggestion
An account-level API token provides API access to your account with k6. To generate the token, go to **Account settings** and select **API token**.
From there you can copy, see, and regenerate the token.
```

![account token view](./images/04-Token/account-api-token-view.png)

## Organization API token

> Organization level API token provides API access to your organization with k6. To generate a token select **API token** menu item under **Organization settings**, where you'll be able to create, see and regenerate the tokens.

Reviewer: ```suggestion
An organization-level API token provides API access to your organization with k6. To generate a token, go to **Organization settings** and select **API token**.
From there you can create, see, and regenerate the tokens.
```

Maximum amount of tokens an organization can create is 5.

![organization token view](./images/04-Token/organization-api-token-view.png)

> #### Access to organization settings
>

> > Only [organization admins](/cloud/project-and-team-management/members/#admin) can access organization settings view.

Reviewer: ```suggestion
> Only [organization admins](/cloud/project-and-team-management/members/#admin) can access organization settings.
```

Below you'll find instructions on how to generate API tokens and examples on how to utilize them.

> #### Google/GitHub Single Sign-On Users
>
> For Single Sign-On (SSO) users, `k6 login cloud` requires a k6 Cloud account email and password. You will need to create a password using [Forgot Password](https://app.k6.io/account/forgot), or you can instead authenticate via API token. See [Authenticating with API token](#authenticating-with-api-token) for more information.

> #### Docker Users
>
> If you're running k6 in a Docker container, make sure that the k6 config file where the k6 Cloud API authentication information is stored is persisted via a Docker volume to the host machine, using the `-c/--config PATH/TO/CONFIG_FILE` CLI flag. Example:
```bash
docker run --rm -i -v /path/on-host:/path/in-container/ grafana/k6 login cloud -c /path/in-container/config.json
```

> #### Integrating with CI
>

> > If you are integrating k6 into your CI pipeline, we recommend using one of the token methods to authenticate and not exposing your username/password within your CI configuration files or as variables.

Author: @ana-yankova do you think it makes sense to add a link here to `Organization API token` section here for visibility purposes?
